Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/loader/impl/external_media_service/ExternalMediaServiceExtraInfoFetcher.kt

```diff
@@ -0,0 +1,50 @@
+package com.github.adamantcheese.chan.core.loader.impl.external_media_service
+
+import android.graphics.Bitmap
+import com.github.adamantcheese.base.ModularResult
+import com.github.adamantcheese.chan.core.loader.impl.post_comment.ExtraLinkInfo
+import io.reactivex.Flowable
+import okhttp3.Response
+
+/**
+ * Base interface for link extra info fetcher. For now only [YoutubeMediaServiceExtraInfoFetcher] is
+ * supported.
+ * */
+internal interface ExternalMediaServiceExtraInfoFetcher {
+    /**
+     * Each fetcher must have it's own type
+     * */
+    val fetcherType: FetcherType
+
+    fun isEnabled(): Boolean
+
+    /**
+     * Icon to prepend the link with
+     * */
+    fun getIconBitmap(): Bitmap
+
+    fun getFromCache(postUid: String, url: String): Flowable<ModularResult<ExtraLinkInfo?>>
+
+    fun storeIntoCache(
+            postUid: String,
+            loadableUid: String,
+            url: String,
+            extraLinkInfo: ExtraLinkInfo
+    ): Flowable<ModularResult<Unit>>
+
+    /**
+     * Whether a link belongs to this fetcher
```

Comment:
"Whether this fetcher can parse the link. A link must only be able to get parsed by one fetcher."